Comp2023/src/main/java/frc/robot/commands/DriveLimelightPath.java

```diff
@@ -86,35 +97,58 @@ public int getSignFromId(int targetId)
     return (targetId <= 4) ? -1 : 1;
   }
 
-  public Pose2d calculateTarget(int targetId, VIConsts.VIGoalDirection goalDirection)
+  public Pose2d getGoalPose(VIConsts.VIGoalDirection goalDirection)
   {
+    int targetId = m_vision.getTargetID( );
+
+    if ((!m_vision.isAprilTagValid(targetId)) || (targetId < 0))
+    {
+      return null;
+    }
+
     Pose2d targetPose = VIConsts.kAprilTagPoses.get(targetId);
     double goalXValue = 0;
     double goalYValue = 0;
     String strName;
 
-    goalXValue = targetPose.getX( ) + getSignFromId(targetId) * VIConsts.kAdjustPathX;
+    if (targetId == 4 || targetId == 5)

```

Comment:
A comment here stating that 4 and 5 are substation IDs would help (or maybe a method isSubstationId().